src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/AbstractApprovalTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2014 Jesse Glick, CloudBees Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.scriptsecurity.scripts;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public abstract class AbstractApprovalTest<T extends Approvable<T>> {
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+
+    /** Creates a new approvable to test. */
+    abstract T create() throws Exception;
+
+    @Test public void noSecurity() throws Exception {
+        create().use();
+    }
+
+    @Test public void withSecurityRejected() throws Exception {
+        configureSecurity();
+        create().assertCannotUse();
+    }
+
+    @Test public void withSecurityAccepted() throws Exception {
+        configureSecurity();
+        create().assertCannotUse().approve().use();
```

Comment:
:ant: Does this not subsume `withSecurityRejected`, increasing test time to no purpose?